File: backend/CCMS.Domain/Entities/Payment.cs
using CCMS.Domain.Enums;

namespace CCMS.Domain.Entities;

public class Payment : BaseEntity
{
    public Guid BookingId { get; set; }
    public Guid UserId { get; set; }

    // Razorpay identifiers
    public string RazorpayOrderId { get; set; } = string.Empty;
    public string? RazorpayPaymentId { get; set; }
    public string? RazorpaySignature { get; set; }

    // Amount
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "INR";

    // Status
    public PaymentStatus Status { get; set; } = PaymentStatus.None;

    // Gateway response (raw JSON from Razorpay)
    public string? GatewayResponse { get; set; }

    // Refund
    public DateTime? RefundedAt { get; set; }
    public decimal? RefundAmount { get; set; }

    // Navigation properties
    public virtual Booking Booking { get; set; } = null!;
    public virtual User User { get; set; } = null!;
}

File: frontend/src/types/payment.ts
export interface Payment {
    id: string;
    bookingId: string;
    userId: string;
    razorpayOrderId: string;
    razorpayPaymentId?: string;
    amount: number;
    currency: string;
    status: string;
    createdAt: string;
    refundedAt?: string;
    refundAmount?: number;
}

export interface CreateOrderResponse {
    orderId: string;
    amount: number;
    currency: string;
    keyId: string;
    bookingId: string;
    virtualAccountNumber?: string;
    virtualAccountIfsc?: string;
    paymentExpiresAt?: string;
}

export interface VerifyPaymentRequest {
    razorpayOrderId: string;
    razorpayPaymentId: string;
    razorpaySignature: string;
    bookingId: string;
}

export interface BookingPaymentStatus {
    bookingId: string;
    paymentStatus: string;
    paymentMethod?: string;
    paymentExpiresAt?: string;
    razorpayOrderId?: string;
}

export interface Payout {
    id: string;
    screenOwnerId: string;
    bookingId?: string;
    type?: string; // Advance, Final, Full
    advancePercentage?: number;
    grossAmount: number;
    commissionPercentage: number;
    commissionAmount: number;
    netAmount: number;
    currency: string;
    status: string;
    periodStart?: string;
    periodEnd?: string;
    adminNotes?: string;
    createdAt: string;
    processedAt?: string;
}

export interface PayoutSummary {
    totalGrossEarnings: number;
    totalCommission: number;
    totalNetEarnings: number;
    pendingPayoutAmount: number;
    completedPayoutAmount: number;
    currency: string;
}

// Razorpay Checkout types
export interface RazorpayOptions {
    key: string;
    amount: number;
    currency: string;
    order_id: string;
    name: string;
    description: string;
    handler: (response: RazorpayResponse) => void;
    prefill?: {
        name?: string;
        email?: string;
        contact?: string;
    };
    theme?: {
        color?: string;
    };
    modal?: {
        ondismiss?: () => void;
    };
}

export interface RazorpayResponse {
    razorpay_order_id: string;
    razorpay_payment_id: string;
    razorpay_signature: string;
}

declare global {
    interface Window {
        Razorpay: new (options: RazorpayOptions) => {
            open: () => void;
            close: () => void;
        };
    }
}
